azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/KeyVaultProcessor.java

```diff
@@ -39,24 +37,11 @@ public void postProcessEnvironment(ConfigurableEnvironment environment, SpringAp
                     environment.getProperty(Constants.AZURE_TOKEN_ACQUIRE_TIMEOUT_IN_SECONDS));
         }
 
-        if (enabled) {
-            final KeyVaultClient kvClient = new KeyVaultClient(
-                    new AzureKeyVaultCredential(clientId, clientKey, timeAcquiringTimeoutInSeconds));
-
-            try {
-                final MutablePropertySources sources = environment.getPropertySources();
-                final KeyVaultOperation kvOperation = new KeyVaultOperation(kvClient, vaultUri);
-
-                if (sources.contains(StandardEnvironment.SYSTEM_ENVIRONMENT_PROPERTY_SOURCE_NAME)) {
-                    sources.addAfter(StandardEnvironment.SYSTEM_ENVIRONMENT_PROPERTY_SOURCE_NAME,
-                            new KeyVaultPropertySource(kvOperation));
-                } else {
-                    sources.addFirst(new KeyVaultPropertySource(kvOperation));
-                }
-
-            } catch (Exception e) {
-                e.printStackTrace();
-            }
+        if (enabled && ClassUtils.isPresent("com.microsoft.azure.keyvault.KeyVaultClient",
+                null)) {
```

Comment:
Use the ClassLoader's Class rather than `null`. I would also wrap that in a separate `private boolean isKeyVaultClientAvailable()`